tests/functional/conftest.py

```diff
@@ -0,0 +1,20 @@
+import subprocess
+
+import pytest
+
+
+@pytest.fixture(scope="session", autouse=True)
+def install_dcgm_snap():
+    """Install the snap and enable dcgm-exporter service for testing."""
+    snap_build_name = "dcgm_*.snap"
+
+    subprocess.run(
+        f"sudo snap install --dangerous {snap_build_name}",
+        check=True,
+        capture_output=True,
+        shell=True,
```

Comment:
Sorry, I forgot that there is a [snap-api ](https://snapcraft.io/docs/snapd-api#heading--snaps)using [unix socket](https://snapcraft.io/docs/using-the-api) that we can use instead of making those subprocess :grimacing:

I'm not completely sure that it's possible to install locally, but I believe it is because there is the `dangerous` flag. If it's not we should open a bug

Reply:
Do you know if we need to use snap-api? Using subprocesses is more straightforward, in my opinion. Besides, other repos use the same approach, so it's kinda a standard

Reply:
I'm fine with just shelling out though ftr there's also https://github.com/canonical/snapd-glib for interacting with snapd from python

Reply:
I prefer avoiding subprocess when possible because API is more robust. In that case that we are not getting any kind of output I'm ok also with sheeling, but it's not ideal. Closing this to not stuck the PR.